java/debugger.jpda.truffle/test/unit/src/org/netbeans/modules/debugger/jpda/truffle/DebugAllBaseTest.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.debugger.jpda.truffle;
+
+import java.io.File;
+import java.net.URL;
+import junit.framework.Test;
+
+import org.netbeans.api.debugger.DebuggerManager;
+import org.netbeans.api.debugger.jpda.JPDADebugger;
+import org.netbeans.api.debugger.jpda.Variable;
+import org.netbeans.modules.debugger.jpda.truffle.breakpoints.TruffleLineBreakpoint;
+import org.netbeans.modules.debugger.jpda.truffle.frames.TruffleStackFrame;
+import org.netbeans.modules.debugger.jpda.truffle.vars.TruffleVariable;
+import org.netbeans.modules.debugger.jpda.truffle.vars.impl.TruffleScope;
+import org.netbeans.modules.javascript2.debug.breakpoints.JSLineBreakpoint;
+
+public class DebugAllBaseTest extends JPDATestCase {
+
+    private static final String SCRIPT_NAME = "DebuggerBase";
+    private static final String[] SCRIPT_EXTENSIONS = { "js", "py", "r", "ruby" };
+    private static final String[] LAUNCHERS = { "js", "graalpython", "Rscript", "ruby" };
+
+    public DebugAllBaseTest(String name) {
+        super(name);
+    }
+
+    public static Test suite() {
+        return createSuite(DebugAllBaseTest.class);
+    }
+
+    private void forAllScripts(ThrowableBiConsumer<String, File> scriptConsumer) {
+        for (int i = 0; i < LAUNCHERS.length; i++) {
+            String launcher = LAUNCHERS[i];
+            File source = new File(sourceRoot, "org/netbeans/modules/debugger/jpda/truffle/scripts/" + SCRIPT_NAME + "." + SCRIPT_EXTENSIONS[i]);
```

Comment:
Shouldn't the path honor `File.separator`?

Reply:
You're right, thanks!